src/station_cmd.cpp

```diff
@@ -2436,7 +2437,7 @@ static CommandCost RemoveAirport(TileIndex tile, DoCommandFlag flags)
 		InvalidateWindowData(WC_STATION_VIEW, st->index, -1);
 
 		if (_settings_game.economy.station_noise_level) {
-			SetWindowDirty(WC_TOWN_VIEW, st->town->index);
+			SetWindowDirty(WC_TOWN_VIEW, nearest->index);
```

Comment:
I would probably actually move this block to the earlier `DC_EXEC` section to avoid having `nearest` spread out over the function. `SetWindowDirty` isn't doing anything by itself, so the location of the call shouldn't matter at all.